tools/NewModuleScript/MainActivityTemplate.kt

```diff
@@ -14,48 +14,53 @@
  *
  */
 
-package com.esri.arcgismaps.sample.displaymap
+package com.esri.arcgismaps.sample.displaycomposablemapview
 
 import android.os.Bundle
-import android.util.Log
-import android.view.View
-import androidx.appcompat.app.AppCompatActivity
-import androidx.databinding.DataBindingUtil
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
 import com.arcgismaps.ApiKey
 import com.arcgismaps.ArcGISEnvironment
-import com.arcgismaps.mapping.ArcGISMap
-import com.arcgismaps.mapping.BasemapStyle
-import com.esri.arcgismaps.sample.displaymap.databinding.ActivityMainBinding
-import com.google.android.material.snackbar.Snackbar
+import com.esri.arcgismaps.sample.displaycomposablemapview.components.ComposeMapView
+import com.esri.arcgismaps.sample.displaycomposablemapview.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
 
-class MainActivity : AppCompatActivity() {
-
-
-
-    // set up data binding for the activity
-    private val activityMainBinding: ActivityMainBinding by lazy {
-        DataBindingUtil.setContentView(this, R.layout.activity_main)
-    }
-
-    private val mapView by lazy {
-        activityMainBinding.mapView
-    }
+class MainActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
         // authentication with an API key or named user is
         // required to access basemaps and other location services
         ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
-        lifecycle.addObserver(mapView)
 
-        // create and add a map with a navigation night basemap style
-        val map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
-        mapView.map = map
+        setContent {
+            SampleAppTheme {
+                SampleApp()
+            }
+        }
     }
 
-    private fun showError(message: String, view: View) {
-        Log.e(localClassName, message)
-        Snackbar.make(view, message, Snackbar.LENGTH_SHORT).show()
+    @Composable
+    private fun SampleApp() {
+        Surface(
+            color = MaterialTheme.colors.background,
+            modifier = Modifier.fillMaxSize(),
+            content = {
+                Column() {
+                    SampleTopAppBar(getString(R.string.app_name))
+                    // composable function that wraps the MapView
+                    ComposeMapView(
+                        modifier = Modifier.fillMaxSize(),
+                        mapViewModel = MapViewModel(application)
+                    )
+                }
+            })
```

Comment:
Two suggestions: 
1) express this as a trailing lambda instead of explicitly setting `content = `. 
2) consider moving this code to a screens package and putting it in a composable called `MapScreen` or `MainScreen`. There is only one screen in this template, but it makes sense to lay out the file structure as if it could grow into a larger app without significant reorganization. The part that is missing is navigation. I think it is ok to omit navigation support since there is one screen, it would be better to implement it anyway. Short of that, I think it is ok to organize the screens so that navigation can be easily brought in.

```suggestion        
             modifier = Modifier.fillMaxSize() {
                MapScreen()
            }
```